lib/lintSource.js

```diff
@@ -251,15 +251,16 @@ function createEmptyPostcssResult(filePath) {
  * There are currently some bugs in the autofixer of Stylelint.
  * The autofixer does not yet adhere to stylelint-disable comments, so if there are disabled
  * ranges we can not autofix this document. More info in issue #2643.
- * Also, if this document is parsed with postcss-jsx and there are nested template
- * literals, it will duplicate some code. More info in issue #4119.
  *
  * @param {PostcssResult} postcssResult
  * @returns {boolean}
  */
 function isFixCompatible({ stylelint }) {
 	// Check for issue #2643
-	if (stylelint.disabledRanges.all.length) return false;
+	const hasUnscopedDisabledRanges = stylelint.disabledRanges.all.length > 0;
+	const hasScopedDisabledRanges = Object.keys(stylelint.disabledRanges).length > 1;
```

Comment:
I think you can turn off autofix for each scoped rule.
Change the scope data to be passed to the third argument of `ruleFunction`.

Reply:
> I think you can turn off autofix for each scoped rule.

Am I right in thinking that would give us a more granular workaround where autofix is turned off (for the entire source):

- for all rules, if the source contains unscoped disable comments
- for only the applicable rules, if the source only contains scoped disable comments

Would you mind pulling down the pull request and showing me what you mean? And is it something we should add to this pull request or a follow-up?

Reply:
Hi @jeddy3.
I want to fix it like this:

https://github.com/stylelint/stylelint/compare/issue-4704..issue-4704-2

Reply:
The color-hex-length rule does not autofix.

https://github.com/stylelint/stylelint/compare/issue-4704..issue-4704-2#diff-01cbf9c2079c6adc0ce42f035fd16a45R245

Reply:
That's a much better approach, thanks!

If possible, can you:

- use `require('common-tags').stripIndent` to indent the `indentation` rule code snippets, like so:

```js
                          .lint({
				code: stripIndent`
                                     /* stylelint-disable color-hex-length */
                                    a {
                                       color: #ffffff;
                                    }
                                `,
```


- add a comment above the following:

```
// Next two conditionals are temporary measures until #2643 is resolved
isFileFixCompatible &&
!postcssResult.stylelint.disabledRanges[ruleName],
```

As their intent isn't immediately clear.

We should then merge your changes into this branch.

Reply:
I fixed it and merged it into this branch.